packages/core/src/common/uri-command-handler.ts

```diff
@@ -47,82 +46,128 @@ export interface MultiUriCommandHandler extends UriCommandHandler<URI[]> {
- * corrupt if the developer does something like:
- * ```ts
- * new UriAwareCommandHandler<URI[]>(selectionService, handler, { multi: false })
- * ```
- */
-export class UriAwareCommandHandler<T extends MaybeArray<URI>> implements CommandHandler {
+export abstract class AbstractUriAwareCommandHandler<T extends MaybeArray<URI>> implements CommandHandler {
 
     constructor(
         protected readonly selectionService: SelectionService,
         protected readonly handler: UriCommandHandler<T>,
-        protected readonly options?: UriAwareCommandHandler.Options
     ) { }
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    protected getUri(...args: any[]): T | undefined {
-        if (args && args[0] instanceof URI) {
-            // @ts-ignore we want to always return URIs
-            return this.isMulti() ? [args[0]] : args[0];
-        }
-        const { selection } = this.selectionService;
-        if (!this.isMulti()) {
-            return UriSelection.getUri(selection) as T;
-        }
-        const uris = UriSelection.getUris(selection);
-        return uris as T;
-    }
+    protected abstract getArgsWithURI(...args: any[]): [T | undefined, ...any[]];
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
     execute(...args: any[]): object | undefined {
-        const uri = this.getUri(...args);
-        return uri ? this.handler.execute(uri, ...args) : undefined;
+        const [uri, ...others] = this.getArgsWithURI(...args);
+        return uri ? this.handler.execute(uri, ...others) : undefined;
     }
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
     isVisible(...args: any[]): boolean {
-        const uri = this.getUri(...args);
+        const [uri, ...others] = this.getArgsWithURI(...args);
         if (uri) {
             if (this.handler.isVisible) {
-                return this.handler.isVisible(uri as T, ...args);
+                return this.handler.isVisible(uri, ...others);
             }
             return true;
         }
         return false;
     }
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
     isEnabled(...args: any[]): boolean {
-        const uri = this.getUri(...args);
+        const [uri, ...others] = this.getArgsWithURI(...args);
         if (uri) {
             if (this.handler.isEnabled) {
-                return this.handler.isEnabled(uri as T, ...args);
+                return this.handler.isEnabled(uri, ...others);
             }
             return true;
         }
         return false;
     }
+}
+
+export class MonoSelectUriCommandHandler extends AbstractUriAwareCommandHandler<URI> {
+    protected getArgsWithURI(...args: any[]): [URI | undefined, ...any[]] {
+        const [maybeURI, ...others] = args;
+        if (maybeURI instanceof URI) {
+            return [maybeURI, ...others];
+        }
+        const { selection } = this.selectionService;
+        const URIFromSelection = UriSelection.getUri(selection);
```

Comment:
nitpick: wrong case and useless variable.

```ts
const UriFromSelection = UriSelection.getUri(this.selectionService.selection);
```

Reply:
Fair point. I've recased and inlined.